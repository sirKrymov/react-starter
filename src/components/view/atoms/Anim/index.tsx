// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

interface IProps {
  duration?: string;
  children: ReactNode;
  delay?: string;
  type?: 'mount' | 'show';
}

export function Anim({
  children,
  duration = '0.4s',
  delay = '0.2s',
  type = 'show'
}: IProps): ReactElement {
  return (
    <div
      style={{
        animationDuration: duration,
        animationFillMode: 'both',
        animationDelay: delay,
        animationName: type,
        opacity: 0
      }}
    >
      {children}
    </div>
  );
}
