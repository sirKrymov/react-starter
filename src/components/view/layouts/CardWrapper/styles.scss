@import '../../../../styles/configs';
@import '../../../../styles/mixins';

.card-wrapper {
  background-color: var(--bgSecondary);
  border-radius: $border-radius;
  box-shadow: $box-shadow-card;
  min-height: 100px;
}
