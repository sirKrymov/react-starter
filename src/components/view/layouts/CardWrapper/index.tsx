// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  className?: string;
  children: ReactNode;
}

export function CardWrapper({ className, children }: IProps): ReactElement {
  const cardWrapperClass = cn(
    {
      'card-wrapper': true
    },
    className
  );

  return <div className={cardWrapperClass}>{children}</div>;
}
