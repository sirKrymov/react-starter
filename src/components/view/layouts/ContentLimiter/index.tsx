// @SETUP: Please don't forget to remove this component or current comment if component is used in project
// @TODO: Add component tests
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  className?: string;
  children: ReactNode;
  fluid?: boolean;
}

export function ContentLimiter({
  className,
  children,
  fluid
}: IProps): ReactElement {
  const contentLimiterTemplateClass = cn(
    {
      'content-limiter--fluid': fluid,
      'content-limiter': true
    },
    className
  );

  return <div className={contentLimiterTemplateClass}>{children}</div>;
}
