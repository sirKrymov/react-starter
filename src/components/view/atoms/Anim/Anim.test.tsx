import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { CardWrapper } from 'components/view/layouts/CardWrapper';
import { Anim } from '../Anim';

describe('<Anim> atom component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(
      <Anim>
        <CardWrapper>Test</CardWrapper>
      </Anim>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
