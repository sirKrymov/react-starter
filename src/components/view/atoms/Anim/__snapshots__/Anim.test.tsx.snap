// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anim> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  style={
    Object {
      "animationDelay": "0.2s",
      "animationDuration": "0.4s",
      "animationFillMode": "both",
      "animationName": "show",
      "opacity": 0,
    }
  }
>
  <CardWrapper>
    Test
  </CardWrapper>
</div>
`;
